src/StackExchange.Redis/Enums/RedisCommand.cs

```diff
@@ -1,236 +1,470 @@
+    GEOSEARCH,
+    GEOSEARCHSTORE,
+
+    GET,
+    GETBIT,
+    GETDEL,
+    GETEX,
+    GETRANGE,
+    GETSET,
+
+    HDEL,
+    HEXISTS,
+    HGET,
+    HGETALL,
+    HINCRBY,
+    HINCRBYFLOAT,
+    HKEYS,
+    HLEN,
+    HMGET,
+    HMSET,
+    HRANDFIELD,
+    HSCAN,
+    HSET,
+    HSETNX,
+    HSTRLEN,
+    HVALS,
+
+    INCR,
+    INCRBY,
+    INCRBYFLOAT,
+    INFO,
+
+    KEYS,
+
+    LASTSAVE,
+    LATENCY,
+    LINDEX,
+    LINSERT,
+    LLEN,
+    LMOVE,
+    LPOP,
+    LPOS,
+    LPUSH,
+    LPUSHX,
+    LRANGE,
+    LREM,
+    LSET,
+    LTRIM,
+
+    MEMORY,
+    MGET,
+    MIGRATE,
+    MONITOR,
+    MOVE,
+    MSET,
+    MSETNX,
+    MULTI,
+
+    OBJECT,
+
+    PERSIST,
+    PEXPIRE,
+    PEXPIREAT,
+    PEXPIRETIME,
+    PFADD,
+    PFCOUNT,
+    PFMERGE,
+    PING,
+    PSETEX,
+    PSUBSCRIBE,
+    PTTL,
+    PUBLISH,
+    PUBSUB,
+    PUNSUBSCRIBE,
+
+    QUIT,
+
+    RANDOMKEY,
+    READONLY,
+    READWRITE,
+    RENAME,
+    RENAMENX,
+    REPLICAOF,
+    RESTORE,
+    ROLE,
+    RPOP,
+    RPOPLPUSH,
+    RPUSH,
+    RPUSHX,
+
+    SADD,
+    SAVE,
+    SCAN,
+    SCARD,
+    SCRIPT,
+    SDIFF,
+    SDIFFSTORE,
+    SELECT,
+    SENTINEL,
+    SET,
+    SETBIT,
+    SETEX,
+    SETNX,
+    SETRANGE,
+    SHUTDOWN,
+    SINTER,
+    SINTERCARD,
+    SINTERSTORE,
+    SISMEMBER,
+    SLAVEOF,
+    SLOWLOG,
+    SMEMBERS,
+    SMISMEMBER,
+    SMOVE,
+    SORT,
+    SPOP,
+    SRANDMEMBER,
+    SREM,
+    STRLEN,
+    SUBSCRIBE,
+    SUNION,
+    SUNIONSTORE,
+    SSCAN,
+    SWAPDB,
+    SYNC,
+
+    TIME,
+    TOUCH,
+    TTL,
+    TYPE,
+
+    UNLINK,
+    UNSUBSCRIBE,
+    UNWATCH,
+
+    WATCH,
+
+    XACK,
+    XADD,
+    XCLAIM,
+    XDEL,
+    XGROUP,
+    XINFO,
+    XLEN,
+    XPENDING,
+    XRANGE,
+    XREAD,
+    XREADGROUP,
+    XREVRANGE,
+    XTRIM,
+
+    ZADD,
+    ZCARD,
+    ZCOUNT,
+    ZDIFF,
+    ZDIFFSTORE,
+    ZINCRBY,
+    ZINTER,
+    ZINTERCARD,
+    ZINTERSTORE,
+    ZLEXCOUNT,
+    ZMSCORE,
+    ZPOPMAX,
+    ZPOPMIN,
+    ZRANDMEMBER,
+    ZRANGE,
+    ZRANGEBYLEX,
+    ZRANGEBYSCORE,
+    ZRANGESTORE,
+    ZRANK,
+    ZREM,
+    ZREMRANGEBYLEX,
+    ZREMRANGEBYRANK,
+    ZREMRANGEBYSCORE,
+    ZREVRANGE,
+    ZREVRANGEBYLEX,
+    ZREVRANGEBYSCORE,
+    ZREVRANK,
+    ZSCAN,
+    ZSCORE,
+    ZUNION,
+    ZUNIONSTORE,
+
+    UNKNOWN,
+}
+
+internal static class RedisCommandExtensions
 {
-    internal enum RedisCommand
+    /// <summary>
+    /// Gets whether a given command can be issued only to a primary, or if any server is eligible.
+    /// </summary>
+    /// <param name="command">The <see cref="RedisCommand"/> to check.</param>
+    /// <returns><see langword="true"/> if the command is primary-only, <see langword="false"/> otherwise.</returns>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0066:Convert switch statement to expression", Justification = "No, it'd be ridiculous.")]
+    public static bool IsPrimaryOnly(this RedisCommand command)
```

Comment:
This probably shouldn't be public.

Reply:
The class is internal ;) (whole enum is really), but yeah no harm in being clearer!

Reply:
To make this clearer for anyone down the road, gave the same love to all other extensions in #2102.